src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -193,25 +331,101 @@ public static FormulaValue SumTable(EvalVisitor runner, SymbolContext symbolCont
         // Max(1,2,3)     
         internal static FormulaValue Max(IRContext irContext, FormulaValue[] args)
         {
-            return RunAggregator(new MaxAgg(), irContext, args);
+            if (irContext.ResultType == FormulaType.Number)
+            {
+                return RunAggregator(new MaxNumberAgg(), irContext, args);
+            }
+            else if (irContext.ResultType == FormulaType.DateTime)
+            {
+                return RunAggregator(new MaxDateTimeAgg(), irContext, args);
+            }
+            else if (irContext.ResultType == FormulaType.Date)
+            {
+                return RunAggregator(new MaxDateAgg(), irContext, args);
+            }
+            else if (irContext.ResultType == FormulaType.Time)
+            {
+                return RunAggregator(new MaxTimeAgg(), irContext, args);
+            }
+            else
+            {
+                return new BlankValue(irContext);
+            }
```

Comment:
Is it possible to hit this? (will binder have prevented).

Reply:
removed